scripts/mobile-flow.js

```diff
@@ -118,10 +113,26 @@ module.exports = function mobileFlow(runner, argv, clientId) {
                             Promise.resolve(updatedRecord),
                             Promise.resolve(doSyncRecordsResult.clientRecs)
                           ]))))
+
           .spread((syncResponse, updatedRecord, clientRecs) => act('Device: sync Complete result', () => create(
             'result',
             makeResult.updateComplete(updatedRecord.data.id, user.id, myWorkorderId),
-            updatedRecord, hashes.result, queryParams(user.id).result, [], 'update')))
+            updatedRecord, hashes.result, queryParams(user.id).result, [], 'update'))
+                  .then(result => doSyncRecords('result', clientRecs[datasets.indexOf('result')], queryParams(user.id).result)
+                        .then(doSyncRecordsResult => Promise.all([
+                          Promise.resolve(result),
+                          Promise.resolve(_.find(
+                            _.get(doSyncRecordsResult, 'res.create', {}),
+                            r => r.data.id === _.map(result.updates.applied, a => a.uid)[0]
```

Comment:
can use `_.find` here to return the first element instead of creating a full new array.